### PR TITLE
Add Support For Istio-Agent

### PR DESCRIPTION
**Pull request by chenyanlann**
## What is the purpose of the change

Related: #10579

Add Support For Istio-Agent

## Brief changelog

Use xds://xxx:xxx?use-agent=true to enable this

Just update the Dubbo-Xds package and add the annotation to 

https://github.com/apache/dubbo-samples/blob/master/dubbo-samples-xds/dubbo-samples-xds-provider/src/main/resources/k8s/Deployment.yml

https://github.com/apache/dubbo-samples/blob/master/dubbo-samples-xds/dubbo-samples-xds-consumer/src/main/resources/k8s/Deployment.yml

Like this

template:
    metadata:
      labels:
        app: dubbo-samples-xds-provider
      annotations:
        inject.istio.io/templates: grpc-agent
        proxy.istio.io/config: '{"holdApplicationUntilProxyStarts": true}

Then follow the instructions in [Dubbo Proxyless](https://dubbo.apache.org/zh/overview/tasks/mesh/proxyless/)

Or you can find more information here https://github.com/chenyanlann/dubbo-samples/tree/master/dubbo-samples-xds

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-xds/pom.xml [hunk 1]
```diff
@@ -50,16 +50,19 @@
         <dependency>
             <groupId>io.grpc</groupId>
             <artifactId>grpc-protobuf</artifactId>
+            <version>1.47.0</version>
```
Comment (AlbumenJ) [inline dubbo-xds/pom.xml:53]: dependency version should be managed within `dubbo-dependencies-bom`

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java [hunk 2]
```diff
@@ -16,26 +16,38 @@
  */
 package org.apache.dubbo.registry.xds.util;
 
+import io.grpc.*;
```
Comment (AlbumenJ) [inline dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java:19]: do not use comma import

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java (hunk 2 above)
Comment (wangchengming666) [inline dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java:19]: 这里最好别用`*`

**Review by wangchengming666 — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java [hunk 3]
```diff
@@ -46,17 +49,30 @@ public class XdsChannel {
...
-            XdsCertificateSigner.CertPair certPair = signer.GenerateCert(url);
-            SslContext context = GrpcSslContexts.forClient()
-                .trustManager(InsecureTrustManagerFactory.INSTANCE)
-                .keyManager(new ByteArrayInputStream(certPair.getPublicKey().getBytes(StandardCharsets.UTF_8)),
-                    new ByteArrayInputStream(certPair.getPrivateKey().getBytes(StandardCharsets.UTF_8)))
-                .build();
-            managedChannel = NettyChannelBuilder.forAddress(url.getHost(), url.getPort()).sslContext(context)
-                .build();
-        } catch (SSLException e) {
+            if(!url.getParameter("useAgent",false)) {
```
Comment (chickenlj) [inline dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java:52]: Please use the dash formatted variable name `use-agent` and define it as a constant.

Also, we need a demo and documentation introducing the usage of this key and the whole bootstrap file.

**Comment by chickenlj** [comment]
Should we unify the configuration of both `agent` and `non-agent` to bootstrap file later?

File: dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java (hunk 3 above)
Comment (chenyanlann) [reply dubbo-xds/src/main/java/org/apache/dubbo/registry/xds/util/XdsChannel.java]: There is more information in https://github.com/apache/dubbo/pull/10567#issue-1364376188

**Review by chickenlj — APPROVED** [review]
LGTM.

**Pull request merged**